src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { useUID } from 'react-uid';
+import styles from './ProgressBar.module.css';
+import Text from '../Text';
+
+export type Props = {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Label associated with naming the progress bar.
+   */
+  label: string;
+  /**
+   * Max value to be represented by the progress bar.
+   */
+  max: number;
+  /**
+   * The amount of segments to be represented by the progress bar.
+   * Recommended max number of segments is 10.
+   */
+  segmentCount: number;
+  /**
+   * The value that each segment represents.
+   */
+  segmentValue: number;
+  /**
+   * An optional unit of measurement to describe the progress.
+   */
+  unit?: string;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {ProgressBar} from "@chanzuckerberg/eds";`
+ *
+ * A progress bar component that indicating the completion progress of a task.
+ */
+export const ProgressBar = ({
+  className,
+  label,
+  segmentCount,
+  segmentValue,
+  max,
+  unit,
+  ...other
+}: Props) => {
+  if (process.env.NODE_ENV !== 'production' && segmentCount > 10) {
+    console.warn('Progress bar segment count should be capped at 10');
+  }
+
+  // Creates an array of length segmentCount and fills them with styled divs to represent the segments.
+  const segments = segmentCount
+    ? Array.from({ length: segmentCount }, (_, index) => (
+        <div
+          className={styles['progress-bar__segment']}
+          key={'progress-bar__segment-' + index}
+          style={{ width: `${(segmentValue / max) * 100}%` }}
```

Comment:
I noticed while playing with the interactive one that there's a little jumpiness when the last segment is completed. I think it's because the width of the segments doesn't take into account the gap between them, so they take up more space when the bar doesn't have all the segments in yet. For example, when the bar is 1200px wide and has max 10 segments, the segments are 120px wide when the bar is not full and 116px wide when the bar is full.

https://user-images.githubusercontent.com/7761701/205736755-f86a056e-d4f2-4af1-809b-ffd27b47b426.mp4

Not blocking but wanted to let you know in case you wanted to try to fix it. I don't know how often users will even see the bar change from being incomplete to complete in front of them, so it might be the kind of thing we notice more when demoing the component.

Reply:
Good callout, will tinker with some css calc to make this look better

Reply:
This took a little more calculation than I had previously anticipated lol:
``` tsx
// Calculates the segment width to prevent jumping once progress bar is full
const segmentsPerBar = Math.ceil(max / segmentValue);
const gapMultiplier = (segmentsPerBar - 1) / segmentsPerBar;
const segmentWidthBeforeGap = (segmentValue / max) * 100;
const segmentWidth = `calc(${segmentWidthBeforeGap}% - var(--eds-size-half) * ${gapMultiplier})`;
return (
  <div
    className={styles['progress-bar__segment']}
    key={'progress-bar__segment-' + index}
    style={{ width: segmentWidth }}
  />
);
```
Let me know if you think this either needs more comments or is overkill.
Also updated the interactive story so the segment value and max can be changed to test this.
@dierat 

https://user-images.githubusercontent.com/86632227/206040757-41103fc4-aefe-4e81-aba0-9091a15cbad3.mov

Reply:
Damn that's nice! That's cool that you got it to figure out a percentage so it doesn't need to recalculate when the progress bar changes width. I would maybe add that to your comment, that it creates a percentage, because I was confused why you were doing it this way until I saw that in the `calc` line.

Reply:
Added another explaining the percentage calc and the subtraction of the gap space